lms/djangoapps/courseware/views/views.py

```diff
@@ -549,7 +549,8 @@ def register_user_access_warning_messages(request, course):
                 )
             )
         else:
-            if not CourseEnrollment.is_enrolled(request.user, course.id) and not allow_anonymous:
+            if not CourseEnrollment.is_enrolled(request.user, course.id) and \
+                    not allow_anonymous and not course.invitation_only:
```

Comment:
Does the condition below this not check if the course is open for self-enrollment? If not, it might make sense to check for `course.invitation_only` along with the condition below (something like `if course_open_for_self_enrollment(course.id) and not course.invitation_only:`, since it has an else statement which shows the appropriate message if the course does not allow self-enrollment.

Reply:
It does not. `course_open_for_self_enrollment(course.id)` checks two things:
- presence of 'CCX' in the course key
- course enrollment dates are open
It does not check the invitation only condition.

Sure thing, I'll add it.

Reply:
Not sure about `if course_open_for_self_enrollment(course.id) and not course.invitation_only`. Would'nt the invitation flag already be considered in the condition above?

Reply:
I've made the change.